packages/angular_devkit/build_angular/src/builders/browser-esbuild/compiler-plugin.ts

```diff
@@ -0,0 +1,411 @@
+      );
+      const lineStart = ts.getPositionOfLineAndCharacter(info.file, line, 0);
+      const lineEnd =
+        line < lastLine
+          ? ts.getPositionOfLineAndCharacter(info.file, line + 1, 0)
+          : info.file.text.length;
+      note.location.lineText = info.file.text.slice(lineStart, lineEnd).trimEnd();
+    }
+  }
+
+  return note;
+}
+
+/**
+ * Converts a TypeScript Diagnostic message into an esbuild compatible message object.
+ * @param diagnostic The TypeScript diagnostic to convert.
+ * @param host A TypeScript FormatDiagnosticsHost instance to use during conversion.
+ * @returns An esbuild diagnostic message as a PartialMessage object
+ */
+function convertTypeScriptDiagnostic(
+  diagnostic: ts.Diagnostic,
+  host: ts.FormatDiagnosticsHost,
+): PartialMessage {
+  let codePrefix = 'TS';
+  let code = `${diagnostic.code}`;
+  if (diagnostic.source === 'ngtsc') {
+    codePrefix = 'NG';
+    // Remove `-99` Angular prefix from diagnostic code
+    code = code.slice(3);
+  }
+
+  const message: PartialMessage = {
+    ...convertTypeScriptDiagnosticInfo(diagnostic, host, `${codePrefix}${code}: `),
+    // Store original diagnostic for reference if needed downstream
+    detail: diagnostic,
+  };
+
+  if (diagnostic.relatedInformation?.length) {
+    message.notes = diagnostic.relatedInformation.map((info) =>
+      convertTypeScriptDiagnosticInfo(info, host),
+    );
+  }
+
+  return message;
+}
+
+// This is a non-watch version of the compiler code from `@ngtools/webpack` augmented for esbuild
+// eslint-disable-next-line max-lines-per-function
+export function createCompilerPlugin(
+  pluginOptions: { sourcemap: boolean; tsconfig: string; advancedOptimizations?: boolean },
+  styleOptions: BundleStylesheetOptions,
+): Plugin {
+  return {
+    name: 'angular-compiler',
+    // eslint-disable-next-line max-lines-per-function
+    async setup(build: PluginBuild): Promise<void> {
+      // This uses a wrapped dynamic import to load `@angular/compiler-cli` which is ESM.
+      // Once TypeScript provides support for retaining dynamic imports this workaround can be dropped.
+      const compilerCli = await loadEsmModule<typeof import('@angular/compiler-cli')>(
+        '@angular/compiler-cli',
+      );
+
+      // Temporary deep import for transformer support
+      const {
+        createAotTransformers,
+        mergeTransformers,
+      } = require('@ngtools/webpack/src/ivy/transformation');
+
+      // Setup defines based on the values provided by the Angular compiler-cli
+      build.initialOptions.define ??= {};
+      for (const [key, value] of Object.entries(compilerCli.GLOBAL_DEFS_FOR_TERSER_WITH_AOT)) {
+        if (key in build.initialOptions.define) {
+          // Skip keys that have been manually provided
+          continue;
+        }
+        // esbuild requires values to be a string (actual strings need to be quoted).
+        // In this case, all provided values are booleans.
```

Comment:
This was more of a note for the future in case someone wanted to introduce a string.  The escaping is something that the provider of the value would need to do since there are two possible string-based scenarios.  One where the string is meant to be a string literal in the resulting code and this is the one that would need to be escaped.  The second is where the string is meant to be an identifier in the resulting code and this should not be escaped.